app/src/main/java/com/kickstarter/ui/activities/NewsletterActivity.kt

```diff
@@ -1,13 +1,83 @@
 package com.kickstarter.ui.activities
 
-import android.support.v7.app.AppCompatActivity
+import android.databinding.DataBindingUtil
 import android.os.Bundle
+import android.support.annotation.NonNull
+import butterknife.BindColor
+import com.jakewharton.rxbinding.view.RxView
 import com.kickstarter.R
+import com.kickstarter.databinding.ActivityNewsletterBinding
+import com.kickstarter.libs.BaseActivity
+import com.kickstarter.libs.Build
+import com.kickstarter.libs.CurrentUserType
+import com.kickstarter.libs.KSString
+import com.kickstarter.libs.qualifiers.RequiresActivityViewModel
+import com.kickstarter.libs.utils.BooleanUtils.isTrue
+import com.kickstarter.libs.utils.SwitchCompatUtils
+import com.kickstarter.libs.utils.ViewUtils
+import com.kickstarter.models.User
+import com.kickstarter.ui.data.Newsletter
+import com.kickstarter.viewmodels.NewsletterViewModel
+import rx.android.schedulers.AndroidSchedulers
 
-class NewsletterActivity : AppCompatActivity() {
+@RequiresActivityViewModel(NewsletterViewModel.ViewModel::class)
+class NewsletterActivity : BaseActivity<NewsletterViewModel.ViewModel>() {
+
+    private lateinit var binding: ActivityNewsletterBinding
+
+    private lateinit var currentUserType: CurrentUserType
+    private lateinit var build: Build
+    private lateinit var ksString: KSString
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_newsletter)
+        binding = DataBindingUtil.setContentView(this, R.layout.activity_newsletter)
+
+        this.build = environment().build()
+        this.currentUserType = environment().currentUser()
+        this.ksString = environment().ksString()
+
+        this.viewModel.outputs.user()
+                .compose(bindToLifecycle())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(this::displayPreferences)
+
+        this.viewModel.errors.unableToSavePreferenceError()
+                .compose(bindToLifecycle())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe { ViewUtils.showToast(this, getString(R.string.profile_settings_error)) }
+
+        this.viewModel.outputs.showOptInPrompt()
+                .compose(bindToLifecycle())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(this::showOptInPrompt)
+
+        binding.apply {
```

Comment:
Instance variables should be referenced by using `this`. Going forward I'll comment "this" where I see it's missing.